javaslang/src/main/java/javaslang/collection/BitSet.java

```diff
@@ -0,0 +1,1024 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import javaslang.Function1;
+import javaslang.Tuple;
+import javaslang.Tuple2;
+import javaslang.Tuple3;
+import javaslang.control.Option;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.function.*;
+import java.util.stream.Collector;
+
+import static javaslang.collection.Comparators.naturalComparator;
+
+/**
+ * An immutable {@code BitSet} implementation.
+ *
+ * @author Ruslan Sennov
+ * @since 2.1.0
+ */
+public interface BitSet<T> extends SortedSet<T> {
+
+    long serialVersionUID = 1L;
+
+    class Builder<T> {
+
+        final static Builder<Integer> DEFAULT = new Builder<>(i -> i, i -> i);
+
+        final Function1<Integer, T> fromInt;
+        final Function1<T, Integer> toInt;
```

Comment:
Both <code>fromInt</code> and <code>toInt</code> fields of <code>AbstractBitSet</code> should be serializable. For example, you can see that <code>BitSetTest.Mapper</code> is serializable class. If we change signatures from <code>Function1</code> to <code>Function</code> these tests (I'll add them to BitSetTest) will fail
```java
    @Test
    public void shouldSerializeDeserializeNativeBitSet() {
        final Object actual = deserialize(serialize(BitSet.of(1, 2, 3)));
        final Object expected = BitSet.of(1, 2, 3);
        assertThat(actual).isEqualTo(expected);
    }

    @Test
    public void shouldSerializeDeserializeEnumBitSet() {
        final Object actual = deserialize(serialize(BitSet.withEnum(E.class).of(E.V1, E.V2)));
        final Object expected = BitSet.withEnum(E.class).of(E.V1, E.V2);
        assertThat(actual).isEqualTo(expected);
    }
```
Or we can introduce new <code>SerializableFunction</code> interface but I'm not sure this is good way